buildSrc/src/main/kotlin/WasmConfig.kt

```diff
@@ -0,0 +1,63 @@
+/*
+ * Copyright 2014-2021 JetBrains s.r.o and contributors. Use of this source code is governed by the Apache 2.0 license.
+ */
+
+import org.gradle.api.*
+import org.gradle.kotlin.dsl.*
+import org.jetbrains.kotlin.gradle.targets.js.dsl.*
+import java.io.*
+
+fun Project.configureWasm() {
+    configureWasmTasks()
+
+    kotlin {
+        sourceSets {
+            val wasmJsTest by getting {
+                dependencies {
+                    implementation(npm("puppeteer", "21.5.0"))
+                }
+            }
+        }
+    }
+
+    configureWasmTestTasks()
+}
+
+private fun Project.configureWasmTasks() {
+    kotlin {
+        @OptIn(ExperimentalWasmDsl::class)
+        wasmJs {
+            nodejs {
+                testTask(
+                    Action {
+                        useMocha {
+                            timeout = "10000"
+                        }
+                    }
+                )
+            }
+
+            browser {
+                testTask(
+                    Action {
+                        useKarma {
+                            version = "3.2.0"
+                            useChromeHeadlessWasmGc()
+                            useConfigDirectory(File(project.rootProject.projectDir, "karma"))
+                        }
+                    }
+                )
+            }
+        }
+    }
+}
+
+private fun Project.configureWasmTestTasks() {
+    val shouldRunWasmBrowserTest = !hasProperty("teamcity") || hasProperty("enable-wasm-tests")
```

Comment:
Please keep the same property name as for js. It looks like we don't need a separate.